app/src/store/data.ts

```diff
@@ -37,9 +37,10 @@ import {
 } from 'src/utils/chains';
 import { DefaultStorage, getStorageKey, setStorageKey } from 'src/utils/data/utils';
 import { GRANT_ROUND_ABI, ERC20_ABI } from 'src/utils/constants';
+import { DataStore } from 'src/types';
```

Comment:
(arbitrary comment location to start a discussion)

If you clear site data, how long does it take for the page to load? For me, with `VITE_DGRANTS_CHAIN_ID=137` it still took about 8-10 seconds which is pretty long

Reply:
took about 5 seconds for me

Reply:
We mentioned this on standup yesterday but I forget what we said about this slowness—I think we said a second follow-up PR will be coming?

Reply:
Takeaways from standup yesterday were:

1. We need to investigate *if* the datastore is actually being called twice (before @gdixon's singleton fix) and if it its, get to the root cause of that issue, rather than the working-but-shouldn't-be-needed singleton approach here
2. We'll do separate PR's to explore other optimizations and improvements to UX, most notably rendering the page once on-chain data is fetched, then lazy loading the metadata content from IPFS into place

Let me know if I missed anything @gdixon.

On this note, has anyone been able to look into point 1. yet?

Reply:
I've looked in to 1 and I'm not able to prove that the singleton approach offers any performance benefits 👍

Reply:
Number 2 is being tracked here: https://github.com/dcgtc/dgrants/issues/510